.github/workflows/main.yml

```diff
@@ -90,18 +102,37 @@ jobs:
     - name: Use strawberry perl
       if: startsWith(matrix.os, 'windows')
       run: echo OPENSSL_SRC_PERL=C:/Strawberry/perl/bin/perl >> $GITHUB_ENV
-    - run: |
+    - name: Run tests (not Windows)
+      if: "!startsWith(matrix.os, 'windows')"
+      run: |
         set -e
         cargo generate-lockfile
         ./ci/run-docker.sh ${{ matrix.target }}
-      if: "!startsWith(matrix.os, 'windows')"
-      name: Run tests (not Windows)
-    - run: |
+    - name: Download nasm.exe (Windows)
+      if: startsWith(matrix.os, 'windows') && matrix.nasm_exe == 'installed'
+      run: |
+        WINNASMVERSION='2.15.05'
+        curl -O https://www.nasm.us/pub/nasm/releasebuilds/${WINNASMVERSION}/win64/nasm-${WINNASMVERSION}-win64.zip
+        unzip nasm-${WINNASMVERSION}-win64.zip
+        mv nasm-${WINNASMVERSION} nasm_win64
+    - name: Set NO_NASM env var (Windows)
+      if: startsWith(matrix.os, 'windows') && matrix.env_no_nasm == 'set'
+      run: echo OPENSSL_RUST_NO_NASM=1 >> $GITHUB_ENV
+    - name: Run tests (Windows)
+      if: startsWith(matrix.os, 'windows')
+      run: |
+        if [ '${{ matrix.nasm_exe }}' = 'installed' ] ; then
+        export PATH=$PWD/nasm_win64:$PATH
+        fi
         cargo test --manifest-path testcrate/Cargo.toml --target ${{ matrix.target }} -vv
         cargo test --manifest-path testcrate/Cargo.toml --target ${{ matrix.target }} --release -vv
         cargo run --release --target ${{ matrix.target }} --manifest-path testcrate/Cargo.toml --features package -vv
-      if: startsWith(matrix.os, 'windows')
-      name: Run tests (Windows)
+    - name: Verify no-nasm option (Windows)
+      if: startsWith(matrix.os, 'windows') && matrix.env_no_nasm == 'set'
+      run: |
+        # if the nasm.exe is disabled, there should be 'no-asm' in the configure command
+        echo 'CI: In output files, there should be no-asm configured:'
+        grep 'no-asm' target/${{ matrix.target }}/*/build/*/output
```

Comment:
I would prefer to avoid doing this since it's an unstable implementation detail of Cargo. As I've mentioned before I think the build script should panic if nasm is explicitly requested but it's not found.

Reply:
I see this output file path in the cargo book. So, I thought it is a stable feature. Now, I'll remove this.
However, on how to switch nasm.exe on and off, do you mean that we should use the [OPENSSL_RUST_USE_NASM](https://github.com/alexcrichton/openssl-src-rs/pull/87#issuecomment-800356341) way?